### PR TITLE
feature: Typescript-angular support single request param

### LINKED ISSUE #4465: [Feature][typescript-angular] useSingleRequestParameter missing on typescript-angular generator
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

Compare
https://openapi-generator.tech/docs/generators/typescript-fetch
and
https://openapi-generator.tech/docs/generators/typescript-angular

The Angular generator does not contain the useSingleRequestParameter

##### openapi-generator version

master

##### Suggest a fix

I can try to implement this myself but I will probably need some hints to get started as I am new to this project

### PR DESCRIPTION
**Pull request by djairhogeuens**
fixes #4465 Add support for singleRequestParameter in typescript-angular generator

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by djairhogeuens** [comment]
@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10) @akehir (2019/07)

Please have a look :)

File: samples/client/petstore/typescript-angular-v2/default/api/pet.service.ts [hunk 1]
```diff
@@ -64,10 +65,11 @@ export class PetService {
     /**
      * 
      * @summary Add a new pet to the store
-     * @param body Pet object that needs to be added to the store
+     
+     * @param pet Pet object that needs to be added to the store
      */
-    public addPet(body: Pet, extraHttpRequestParams?: RequestOptionsArgs): Observable<{}> {
-        return this.addPetWithHttpInfo(body, extraHttpRequestParams)
+    public addPet(pet: Pet, extraHttpRequestParams?: RequestOptionsArgs): Observable<{}> {
```
Comment (macjohnny) [inline samples/client/petstore/typescript-angular-v2/default/api/pet.service.ts]: please don't run the `bin/openapi3` scripts

**Comment by macjohnny** [comment]
thanks for the PR, I will have a closer look soon

**Comment by macjohnny** [comment]
please add a configuration to https://github.com/OpenAPITools/openapi-generator/blob/master/bin/typescript-angular-petstore-all.sh and https://github.com/OpenAPITools/openapi-generator/blob/master/bin/windows/typescript-angular-petstore-all.bat

**Comment by macjohnny** [comment]
@djairhogeuens can you please commit the executable flag of the newly added script in /bin?

**Comment by djairhogeuens** [comment]
> @djairhogeuens can you please commit the executable flag of the newly added script in /bin?

The executable flag was set, I did notice a small mistake in the script itself when comparing it to the windows version. I ran them both locally and they seem to work fine so I commited that fix and will see if CI passes now.

**Comment by macjohnny** [comment]
the executable flag is still missing:

https://github.com/OpenAPITools/openapi-generator/blob/4083448afd40c84e2c2784e5e929f679be4f3b6f/bin/typescript-angular-v8-petstore-single-request-parameter.sh

see https://github.com/OpenAPITools/openapi-generator/blob/4083448afd40c84e2c2784e5e929f679be4f3b6f/bin/typescript-angular-petstore-all.sh for reference

**Comment by djairhogeuens** [comment]
Hm strange... In my terminal it showed that the file was executable (and I could execute it) but when checking git ls-files --stage it said that it was not executable. I managed to fix it now :)

**Comment by filipvh** [comment]
@macjohnny @djairhogeuens Builds seems to have timed out?

**Comment by macjohnny** [comment]
@filipvh I restarted the build

**Comment by macjohnny** [comment]
@djairhogeuens the CircleCI tests fail due to the docs-page not being up to date. can you please fix this?

**Comment by macjohnny** [comment]
Please also update the following file: https://github.com/openapitools/openapi-generator/blob/master/modules/openapi-generator/src/test/java/org/openapitools/codegen/options/TypeScriptAngularClientOptionsProvider.java

**Comment by djairhogeuens** [comment]
Docs were updated but apparently the order of the parameters matters, fixed it :)
The TypeScriptAngularClientOptionsProvider.java was already updated with the new option

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by djairhogeuens** [comment]
@wing328 When checking in SourceTree it seems correct to me?
image

**Comment by wing328** [comment]
@djairhogeuens thanks for the PR, which has been included in the v4.2.1 release: https://twitter.com/oas_generator/status/1195339336922759168

**Comment by wing328** [comment]
@djairhogeuens please have a look at https://github.com/OpenAPITools/openapi-generator/pull/4479/commits and none of the commits is linked to your Github account.

**Pull request merged**